### ISSUE TITLE
[Suggestion] Buildable Robot Assistants

### ISSUE THREAD

**Issue by Asnro** [tsunamayo/Starship-EVO#5295]
This is something that recently came to my mind when thinking about Starship EVO's past and upcoming features; namely, mechs and crew NPCs.

Why not combine the two? Build your own crew!

Robots would be built in a similar way to handheld guns: place a robot starter block on an item stand and go to town with the regular bricks. Robots would have a few special pieces: a hand, a foot, and an Inverse Kinematics controller that can be used to define joint chains (ie hand>elbow>shoulder). Robots could be any size, but fitting through doorways would be a soft cap on their bulk.

Once built, these robots could do "basic" tasks for the player: ship repair, refueling, mining, even combat with guns attached. To prevent excessive spam the player would need to acquire a software licence for each robot, similar to hiring new living crew, which would handle more "complex" tasks like manning stations amd piloting ships.

This mechanic could help the game stand out even more from its kind, where NPC crews are rare but *custom built* NPC crews are even rarer. Not to mention the whole new world of building and customization available to players with this new dimension. I would love to see personalized amd unique robots populating the galaxy!

Thank you for your time reading this, I hope to hear thoughts from anyone!

**Comment by BigBadKangaroo** [comment]
I really like this idea!

**Comment by Crimson-Artist** [comment]
if you look at the old videos of mechs they already had a system built in to designate joints. The biggest question is AI

**Comment by Majikmonster** [comment]
I like this idea for 3 reasons:

1. You can acquire NPC help before advancing to exoplanetary travel, making early gameplay more adventurous, as well as being able to fill gaps in your crew when organics aren't readily available.
2. The choice between RNG of a uniquely skilled NPC (per Tsuna's idea of crew) or a predictable and consistent (and replaceable) labor force made through a different investment of resources.
3. Robots are cool 🤖

**Comment by tsunamayo** [comment]
@Asnro Yes I really want to do that, one day hopefully! I really need to ship survival first though... Cheers
